pypyr/dsl.py

```diff
@@ -606,7 +619,15 @@ def run_conditional_decorators(self, context):
                             logger.error(
                                 "Error while running step %s", self.name
                             )
-                        raise exc_info
+
+                        if swallow_exc_info is not None:
```

Comment:
so sorry to do this to you. . . because I know it was my comment that set you on this path 😬 Failure on my part to think it through carefully, sorry!

but seeing this in action, it looks like extra complexity that's not actually adding all that much useful to what we're trying to achieve - I forgot that the `save_error` call needed the `swallow_me` result. And now we're going to have an errors list/stack that says `swallow` was True when in fact it wasn't necessarily True, it might just have had a formatting error.

If we instead just had `swallow_me = context.get_formatted_as_type(self.swallow_me, out_type=bool)` on ln 582, without any of the extra try/except catching + re-raising later, this would just raise the formatting exception and break out of the step execution without treating it as a runtime error (i.e it won't save the original exception with `self.save_error()`.

But now that I reconsider, that's probably closer to the desired behaviour - a pipeline formatting issue should be terminal. What do you think?

Reply:
Yes yes, I totally agree that it is overhead handling the swallow formatting error. I have rolled back to the previous, and now the formatting issue is terminal in case of the exception.

Reply:
thanks for your patience! 🙌